astropy/units/format/fits.py

```diff
@@ -10,6 +10,7 @@
 
 import numpy as np
 
+from astropy.units.errors import UnitScaleError
```

Comment:
Feels like this should be `from .. import UnitScaleError` since it is local to the subpackage (or, if not, `from astropy.units import UnitScaleError`). But obviously not a big deal at all

Reply:
Separately, how about also importing `CompositeUnit` so that the import of `core` can be removed?

Reply:
Using relative imports with `..` would violate Ruff rule [TID252 (relative-imports)](https://docs.astral.sh/ruff/rules/relative-imports/), so discussing that is beyond the scope here.

I have no intention of touching anything related to importing things from `astropy.units.core` anywhere in `astropy.units.format` in this pull request:
https://github.com/astropy/astropy/blob/814325422234527424bdc3b5942484cd776342ce/astropy/units/format/__init__.py#L16-L19

Reply:
Ouch, I had forgotten about that! Better stick with your really nice improvements here then!